tests/tests/test_IO/test_cell_io.py

```diff
@@ -55,91 +48,210 @@ def z_vals(data_path):
 
 
 @pytest.fixture
-def cubes_cells():
+def cells_with_artifacts():
     return [
-        Cell([340, 1004, 15], 1),
-        Cell([340, 1004, 15], 1),
-        Cell([392, 522, 10], 1),
+        Cell([340, 1004, 15], -1),
+        Cell([340, 1004, 15], -1),
         Cell([392, 522, 10], 1),
+        Cell([392, 522, 10], 2),
+        Cell([392, 522, 10], 2),
     ]
 
 
-@pytest.fixture
-def roi_sorter_cells():
-    return [
-        Cell([4056, 564, 358], 1),
-        Cell([3989, 267, 570], 1),
-        Cell([4351, 735, 439], 1),
-        Cell([4395, 677, 367], 1),
-    ]
-
-
-def test_get_cells(
-    cubes_cells,
-    roi_sorter_cells,
-    xml_path,
-    yml_path,
-    cubes_dir,
-    roi_sorter_output_dir,
-):
+def test_get_cells_xml(xml_path):
+    """
+    Test that cells can be read from an xml file.
+    """
     cells = cell_io.get_cells(xml_path)
     assert len(cells) == 65
     assert Cell([2536, 523, 1286], 1) == cells[64]
 
+
+@pytest.mark.parametrize("cells_only", [True, False])
+def test_get_cells_xml_cells_only(cells_only, tmp_path, cells_with_artifacts):
+    """
+    Test that cells not of type Cell.CELL (type 2) are correctly removed or
+    kept when reading from xml with the cells_only option.
+    """
+    tmp_cells_out_path = tmp_path / "cells.xml"
+    cell_io.cells_to_xml(
+        cells_with_artifacts, tmp_cells_out_path, artifact_keep=True
+    )
+    cells = cell_io.get_cells(str(tmp_cells_out_path), cells_only=cells_only)
+
+    if cells_only:
+        assert len(cells) == 2
+        for cell in cells:
+            assert cell.type == 2
+    else:
+        assert len(cells) == len(cells_with_artifacts)
+
+
+def test_get_cells_yml(yml_path):
+    """
+    Test that cells can be read from a yml file.
+    """
     cells = cell_io.get_cells(yml_path)
     assert len(cells) == 250
     assert Cell([9170, 2537, 311], 1) == cells[194]
 
-    cells = cell_io.get_cells(cubes_dir)
-    assert len(cells) == 4
-    assert natsorted(cubes_cells) == natsorted(cells)
 
-    cells = cell_io.get_cells(roi_sorter_output_dir)
+@pytest.mark.parametrize(
+    "cells_dir, expected_cells",
+    [
+        (
+            Path("cube_extract") / "cubes",
+            [
+                Cell([340, 1004, 15], 1),
+                Cell([340, 1004, 15], 1),
+                Cell([392, 522, 10], 1),
+                Cell([392, 522, 10], 1),
```

Comment:
These numbers (barring the -1's) are the same as the old fixture above. Is there any way to maybe combine these into one parameterized fixture, as opposed to repeating ourselves a few times?

If that's going to be a pain to implement then maybe it's not worth the effort. I was thinking a fixture that returns a selection of pre-defined cells that we need for a given test?

```python
@pytest.fixture
def selection_of_cells(indices_we_want):
    all_the_cells = [
        Cell(...), Cell(...), ...
    ]
    return all_the_cells[indices_we_want]
```

But that might be more effort than this is worth.

Reply:
I think I could combine these fixtures together, but I wonder if it would make it less clear that 
``` 
[
      Cell([340, 1004, 15], 1),
      Cell([340, 1004, 15], 1),
      Cell([392, 522, 10], 1),
      Cell([392, 522, 10], 1),
 ]
 ```
is the contents of the test data under `tests/data/cube_extract/cubes`, while the other containing -1 values is just arbitrary testing data? I.e. that one should never be modified as it matches written files in the data dir?

Perhaps I could change the fixture with -1 values to the values below to make it clear that only the type is important?
```
@pytest.fixture
def cells_with_artifacts():
    return [
        Cell([1, 2, 3], -1),
        Cell([4, 5, 6], -1),
        Cell([7, 8, 9], 1),
        Cell([10, 11, 12], 2),
        Cell([13, 14, 15], 2),
    ]
```

Reply:
> is the contents of the test data under tests/data/cube_extract/cubes, while the other containing -1 values is just arbitrary testing data

No that's a fair point, I didn't realise that one was being compared against actual reference data due to the GitHub diff cutting it off. I think leave as it is, we can always re-evaluate later.